app/models/split.rb

```diff
@@ -62,15 +64,15 @@ def reweight!(weighting_registry)
   end
 
   def assignment_count_for_variant(variant)
-    assignments.where(variant: variant).count(:id)
+    Assignment.from(assignments.for_presentation, :assignments).where(variant: variant).count(:id)
   end
 
   def build_decision(params = {})
-    Decision.new({ split: self }.merge(params))
+    Decision.new(params.merge(split: self))
```

Comment:
This makes split not overrideable, which seems more secure.

Reply:
Sometimes the key order matters, but that doesn't seem to be the case here, nor would this be (as you said) the most secure way to solve the ordering problem.